Library/ViewModels/LiveStreamDiscoveryViewModel.swift

```diff
@@ -0,0 +1,94 @@
+import KsApi
+import LiveStream
+import Prelude
+import ReactiveSwift
+import Result
+
+public protocol LiveStreamDiscoveryViewModelInputs {
+  func isActive(_ active: Bool)
+  func tapped(liveStreamEvent: LiveStreamEvent)
+  func viewDidLoad()
+}
+
+public protocol LiveStreamDiscoveryViewModelOutputs {
+  var goToLiveStreamContainer: Signal<(Project, Project.LiveStream, LiveStreamEvent), NoError> { get }
+  var goToLiveStreamCountdown: Signal<(Project, Project.LiveStream, LiveStreamEvent), NoError> { get }
+  var loadDataSource: Signal<[LiveStreamEvent], NoError> { get }
+}
+
+public protocol LiveStreamDiscoveryViewModelType {
+  var inputs: LiveStreamDiscoveryViewModelInputs { get }
+  var outputs: LiveStreamDiscoveryViewModelOutputs { get }
+}
+
+public final class LiveStreamDiscoveryViewModel: LiveStreamDiscoveryViewModelType,
+LiveStreamDiscoveryViewModelInputs, LiveStreamDiscoveryViewModelOutputs {
+
+  public init() {
+
+    let projectAndTappedLiveStreamEvent = self.tappedLiveStreamEventProperty.signal.skipNil()
+      .switchMap(freshProjectAndStreamAndEvent(fromLiveStreamEvent:))
```

Comment:
this isn't ideal, but here's the sitch...

from live stream discovery we do not have enough information to take the user straight to the countdown/livestream. for one we don't have the project, and for two we don't have _all_ of the stream event details. 

so, when the user taps we perform two api requests (in parallel) to get that data :/ that means there is a noticeable delay from when you tap to when we do something. we are investigating ways to make this better, and ideally not even necessary to do the api requests.